plugins/modules/idrac_server_config_profile.py

```diff
@@ -816,32 +820,22 @@ def import_scp_redfish(module, idrac, http_share):
             "import_buffer": buffer_text, "target": scp_targets, "share": share_dict, "job_wait": module.params["job_wait"],
             "host_powerstate": module.params["end_host_power_state"], "shutdown_type": module.params["shutdown_type"]
         }
-        scp_response = idrac_import_scp_share(module, idrac, idrac_import_scp_params)
+        scp_response = idrac.import_scp_share(**idrac_import_scp_params)
+        wait_for_job_tracking_redfish(module, idrac, scp_response)
     else:
         scp_response = idrac.import_scp(import_buffer=import_buffer, target=scp_targets, job_wait=module.params["job_wait"])
     scp_response = response_format_change(scp_response, module.params, share.get("file_name"))
     exit_on_failure(module, scp_response, command)
     return scp_response
 
 
-def idrac_import_scp_share(module, idrac, job_params):
-    scp_response = idrac.import_scp_share(**job_params)
-    job_id = scp_response.headers["Location"].split("/")[-1]
-    if module.params["job_wait"]:
-        job_failed, _msg, job_dict, _wait_time = idrac_redfish_job_tracking(
-            idrac, iDRAC_JOB_URI.format(job_id=job_id))
-        if job_failed:
-            module.exit_json(failed=True, status_msg=job_dict, job_id=job_id, msg=FAIL_MSG.format(module.params["command"]))
-        scp_response = job_dict
-    return scp_response
-
-
 def wait_for_job_tracking_redfish(module, idrac, scp_response):
     job_id = scp_response.headers["Location"].split("/")[-1]
     if module.params["job_wait"]:
         job_failed, _msg, job_dict, _wait_time = idrac_redfish_job_tracking(
             idrac, iDRAC_JOB_URI.format(job_id=job_id))
-        if job_failed or "SYS045" in job_dict.get("MessageId", ""):
+        error_codes = ["SYS045", "SYS046"]
```

Comment:
Lets move this to constant, Also lets check if we need to add any error code for preview operation.

Reply:
done